modules/mobile/spec/request/payment_history_request_spec.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require_relative '../support/iam_session_helper'
+require_relative '../support/matchers/json_schema_matcher'
+
+RSpec.describe 'payment_history', type: :request do
+  include JsonSchemaMatchers
+
+  before(:all) do
+    @original_cassette_dir = VCR.configure(&:cassette_library_dir)
+    VCR.configure { |c| c.cassette_library_dir = 'modules/mobile/spec/support/vcr_cassettes' }
+  end
+
+  after(:all) { VCR.configure { |c| c.cassette_library_dir = @original_cassette_dir } }
+
+  before { iam_sign_in }
+
+  describe 'GET /mobile/v0/payment-history' do
+    context 'with successful response' do
+      before do
+        VCR.use_cassette('payment_history/retrieve_payment_summary_with_bdn', match_requests_on: %i[method uri]) do
+          get '/mobile/v0/payment-history', headers: iam_headers, params: nil
+        end
+      end
+
+      it 'returns 200' do
+        expect(response).to have_http_status(:ok)
+      end
+
+      it 'matches expected schema' do
+        expect(response.body).to match_json_schema('payment_history')
+      end
+
+      it 'includes the expected properties for payment history' do
```

Comment:
It might also be good to test the case where `mask_account_number` returns nil.

Reply:
Cool! I updated specs